src/io.c

```diff
@@ -1777,17 +1782,37 @@ _dispatch_stream_cleanup_operations(dispatch_stream_t stream,
 	dispatch_operation_t op, tmp;
 	typeof(*stream->operations) *operations;
 	operations = &stream->operations[DISPATCH_IO_RANDOM];
+#ifdef __LINUX_PORT_HDD__
+	LINUX_PORT_ERROR();
+        // NOT CORRECT (doesn't support removal during iteration)
+	TAILQ_FOREACH(op, operations, operation_list) {
```

Comment:
or at least a header that adds the missing macros, I really don't like the duplication of all of the `FOREACH_SAFE` blocks, that will be a maintenance and merging headache